### ISSUE TITLE
Disapearing layers 

### ISSUE THREAD

**Issue by ghuotvez** [NRCan/GSC-Field-Application#174]
### Reproduce Recipe No. 1
- When taking a photo from quick icons in map page.
- Take a photo with current device.
- Navigates to note page
- Navigate to map page, all layers disappeared...

Notes, photo can be changed for sample and it will still happen.

**Comment by ghuotvez** [comment]
More info in #176 and it seems to happen when the map page is somewhere else then current gps location, let's say 1000km away, then when map page is trying to reposition map page on gps location, selecting field notes then map page clears out the whole view.

### Reproduce

- Navigate to map page
- Pan away from current location
- When map page starts to move back to location select field notes
- Navigate back top map page, all layers should have disapeared.

**Comment by ghuotvez** [comment]
This one is a deep problem coming from who knows where. Here's all the avenues and searches that I've conducted.

#### Checked
- I've check the map object before and after it's happenning and can't see a difference.
- I've checked if it was a layer ordering happening with the grid in xaml
- I've checked databases read/writes and made sure that all db calls are closed properly.
- I've checked async calls and fixed possible problems.
- I've simplified navigation to map page view with less calls.
- I've removed the background methods that customize esri map background with white and green meridian instead of grey default one. Doesn't fix anything.
- I've tried different combination of GPS refresh rate and map viewing point changes on every new location. Basically the rendering of the new location was changed to faster, same or longer then GPS refresh rate.
- I've checked and tested colors of all graphics and controls on map page to make sure nothing gets over the map for some reasons.

#### Possible path
- It does happen more easily on the tablet then desktop, which kind of points to memory usage or memory management.

**Comment by ghuotvez** [comment]
#### Reproduce version 2

-  Create automatically over 300 photo records
-  Navigate back to map page
-  Everything has disappeared.

**Comment by ghuotvez** [comment]
Last changes did fix the reproduce recipe no.2 but still happens when testing on a tablet.

**Comment by ghuotvez** [comment]
Also, I did test a 1000 station field database with last fix and it works fine. So it did fix #162 .

**Comment by ghuotvez** [comment]
Sometimes navigating from notes to map, with a brand new field book in which only a sample was taken, will crash the whole app...

**Comment by ghuotvez** [comment]
#### Checked

- Reverted ESRI sdk back to 10.3.x

**Comment by jameljoseph** [comment]
I am on it Gab!!

**Comment by ghuotvez** [comment]
Sometimes I can see the watermark regarding no layers are available on the background that becomes white, even though it's now grey like esri default. It seems like the map object is losing reference to any layers in there for some reasons. When looking at it, within the object, everything is in there and visible. For testing I removed the watermark and I still get the issue but now without the watermark, so it's not a graphic that is being pushed on top of the map for some reasons.

**Comment by ghuotvez** [comment]
Only happens when location is moving and/or GPS is activated...

**Comment by ghuotvez** [comment]
On the geolocator object, modifying the movement threshold to 1 m gives the me problem almost instantly. Changing the time interval to 2 seconds, seems to fix the issue, but more testing is needed, and 2 seconds interval might be too long if user is in a helicopter

**Comment by ghuotvez** [comment]
@jameljoseph I'm able to reproduce the problem when enabling tablet mode on laptop. There seems to be something going on with it as opposed to desktop version of windows for some reasons.

**Comment by ghuotvez** [comment]
I get this when I crash the whole thing in tablet mode:

`ucrtbase.dll An invalid parameter was passed to a function that considers invalid parameters fatal.`

**Comment by ghuotvez** [comment]
@jameljoseph I removed a VS C++ 2015 reference and it seems to be working fine. Will need more testing on a tablet.

**Comment by jameljoseph** [comment]
How in the world did you figure that out?  I didnt even try to remove references!!

**Comment by ghuotvez** [comment]
Well, still getting the same error on the tablet. We'll need more testing and thinking about this...

**Comment by ghuotvez** [comment]
I've conducted more testing, and I tried the last 3 previous versions of the app (1.2, 2.0, 2.1) and they all have the same glitch/problem.

### Reproduce recipe No. 4 (laptop)

- Activate tablet mode
- Plug in a bluetooth GPS and pair
- Take a station
- Navigate back to map page
- Take a sample
- Navigate back to map page
- Then go to field notes
- Back to map page and everything is gone.

**Comment by ghuotvez** [comment]
@jameljoseph I think I got it for real this time.  This line was the problem: https://github.com/NRCan/GSC-Field-Application/blob/6bbd1aff132b4fff3de600c15ec10e0150e69349/GSCFieldApp/ViewModels/MapPageViewModel.cs#L1271 

So each time the event of a new position is triggered it sets a new graphic of current location and was also moving the map center to current gps position (SetViewPoint). I'm not sure I understand why it was busting the mapview between app page navigation though. I've place the command within it's own little button on the map page and removed it from the new position event. I've tested it on a tablet, but we need more testing in real time and also if the button is a good addition too.
